interfaces/builtin/iio.go

```diff
@@ -0,0 +1,144 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/snapcore/snapd/interfaces"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var iioConnectedPlugAppArmor = []byte(`
+# Description: Give access to a specific IIO device on the system.
+
+###IIO_DEVICE_PATH### rw,
+/sys/bus/iio/devices/###IIO_DEVICE_NAME###/{,**} rw,
+`)
+
+// The type for iio interface
+type IioInterface struct{}
+
+// Getter for the name of the iio interface
+func (iface *IioInterface) Name() string {
+	return "iio"
+}
+
+func (iface *IioInterface) String() string {
+	return iface.Name()
+}
+
+// Pattern to match allowed iio device nodes. It is gonna be used to check the
+// validity of the path attributes in case the udev is not used for
+// identification
+var iioControlDeviceNodePattern = regexp.MustCompile("^/dev/iio:device[0-9]+$")
+
+// Check validity of the defined slot
+func (iface *IioInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	// Does it have right type?
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("slot is not of interface %q", iface))
+	}
+
+	// Creation of the slot of this type
+	// is allowed only by a gadget snap
+	if !(slot.Snap.Type == "gadget" || slot.Snap.Type == "os") {
+		return fmt.Errorf("%s slots only allowed on gadget or core snaps", iface.Name())
+	}
+
+	// Validate the path
+	path, ok := slot.Attrs["path"].(string)
+	if !ok || path == "" {
```

Comment:
@jhodapp Probably a C/C++ habit :) Adding the () will make go fmt unhappy, it will remove them straight away. Try for yourself: https://play.golang.org/p/DUt-Mq13SX (and we require a go fmt "clean" source).